client/header/activity-panel/panels/inbox.js

```diff
@@ -17,10 +17,18 @@ import ActivityHeader from '../activity-header';
 import { EmptyContent, Section } from '@woocommerce/components';
 import sanitizeHTML from 'lib/sanitize-html';
 import { QUERY_DEFAULTS } from 'wc-api/constants';
+import withSelect from 'wc-api/with-select';
 
 class InboxPanel extends Component {
+	componentWillUnmount() {
```

Comment:
Just curious, why `componentWillUnmount` instead of `componentDidMount`? If a note is created while the user has the panel open, it will not be displayed to the user but the `activity_panel_inbox_last_read` date will be assigned to a later date than the note creation date, right?

Reply:
This tradeoff was needed to make it stateless (unless you have another idea here) in https://github.com/woocommerce/woocommerce-admin/pull/1823/commits/7786711e2434687749635976dca82bf674d636ec.

If we update in `componentDidMount` then the read time will always be greater than notes since we are comparing directly against the prop we just updated.

Reply:
Oh, I understand it now. That's a tricky issue.

I guess we could store the time the component is created in the `constructor` or `componentDidMount` and wait to call `updateCurrentUserData` in `componentWillUnmount`. Something like this:

```ES6
class InboxPanel extends Component {
	constructor() {
		this.mountTime = Date.now();
	}
	componentWillUnmount() {
		const userDataFields = {
			[ 'activity_panel_inbox_last_read' ]: this.mountTime,
		};
		this.props.updateCurrentUserData( userDataFields );
	}
	...
```